File: app/src/main/java/cjx/com/diary/view/activity/SplashActivity.java
package cjx.com.diary.view.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.Button;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import cjx.com.diary.R;
import cjx.com.diary.base.BaseActivity;
import cjx.com.diary.presenter.impl.SplashPresenterImpl;
import cjx.com.diary.view.SplashView;

/**
 * Created by bear on 2017/4/17.
 */

public class SplashActivity extends BaseActivity implements SplashView {
    @BindView(R.id.btn_skip)
    Button mSkipBtn;

    private SplashPresenterImpl mSplashPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        ButterKnife.bind(this);
        mSplashPresenter = new SplashPresenterImpl();
        mSplashPresenter.bindView(SplashActivity.this, null);
        mSplashPresenter.setData();
    }

    @OnClick(R.id.btn_skip)
    public void onViewClicked() {
        jumpToMainActivity();
    }

    @Override
    public void jumpToMainActivity() {
        MainActivity.action(mActivity);
        finish();
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        mSplashPresenter.clearTimer();
    }
}

File: app/src/main/java/cjx/com/diary/view/activity/SplashActivity.kt
package cjx.com.diary.view.activity

import android.os.Bundle
import cjx.com.diary.R
import cjx.com.diary.base.BaseActivity
import cjx.com.diary.presenter.impl.SplashPresenterImpl
import cjx.com.diary.view.SplashView
import kotlinx.android.synthetic.main.activity_splash.*

/**
 * description:
 * author: bear .
 * Created date:  2019-06-04.
 */
class SplashActivity : BaseActivity(), SplashView {

    private val mSplashPresenter: SplashPresenterImpl by lazy {
        SplashPresenterImpl()
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        mSplashPresenter?.bindView(this, null)
        mSplashPresenter?.setData()
        btn_skip.setOnClickListener { jumpToMainActivity() }
    }


    override fun jumpToMainActivity() {
        MainActivity.action(this)
        finish()
    }

    override fun onDestroy() {
        super.onDestroy()
        mSplashPresenter?.clearTimer()
    }
}